runtime/bcverify/vrfyhelp.c

```diff
@@ -1218,3 +1229,32 @@ storeVerifyErrorData (J9BytecodeVerificationData * verifyData, I_16 errorDetailC
 	 */
 	liveStack->pc =  (UDATA)currentPC;
 }
+
+/**
+ * @brief Determine whether the specified field only belongs to the current ROM class.
+ *
+ * @param romClass a pointer to J9ROMClass
+ * @param field a pointer to J9ROMFieldRef
+ * @return TRUE if it exists in the current ROM class rather than superclasses
+ *         FLASE when the field doesn't exist
+ */
+static IDATA findFieldFromCurrentRomClass(J9ROMClass *romClass, J9ROMFieldRef *field)
+{
+	J9UTF8 *searchName = J9ROMNAMEANDSIGNATURE_NAME(J9ROMFIELDREF_NAMEANDSIGNATURE(field));
```

Comment:
Should this code be matching the signature as well? I believe it's legal two have two fields of the same name in a class with different signatures.

Reply:
The java compiler will treat it as invalid due to the duplicate field names with different signatures. e.g.
```
public class Test {

public int a;
public float a;

    Test() {
        this.a = 1;
        this.a = 0.1;
    }

Test.java:4: error: variable a is already defined in class Test
public  float a;
              ^
Test.java:9: error: incompatible types: possible lossy conversion from double to int
        this.a = 0.1;
                 ^
2 errors
```
But I believe the compilation can be bypassed with the bytecode manipulation tools (e.g.ASM) by modifying the generated bytecode. So I agreed to match the field signature to avoid such case.